src/git.ts

```diff
@@ -31,26 +31,24 @@ export class VSCodeGit {
 
   waitForRepo(times: number): Promise<{ name: string; owner: string }> {
     let count = 0
-    const timeToStop = count === times
 
     return new Promise((resolve, reject) => {
-      const checkRepoExists = setInterval(() => {
+      let interval = setInterval(() => {
         try {
           const remotes = this.repository._repository.remotes
           if (remotes.length > 0) {
             const remote = remotes[0]
             const parsed = GitUrlParse(remote.pushUrl)
             resolve({ name: parsed.name, owner: parsed.owner })
           } else {
-            if (timeToStop) {
+            if (count === times) {
+              clearInterval(interval)
               reject(new Error("Couldn't get repo details"))
             }
             count++
           }
         } catch (e) {
           reject(new Error("Couldn't get repo details"))
-        } finally {
-          clearInterval(checkRepoExists)
```

Comment:
also WHY did I do this? This immediately cleared the interval, causing this promise to hang forever